src/app/Library/CrudPanel/Traits/Create.php

```diff
@@ -66,11 +66,18 @@ public function getRelationFields()
                 array_push($relationFields, $field);
             }
 
-            if (isset($field['subfields']) &&
+            // if a field has an array name AND subfields
+            // then take those fields into account (check if they have relationships);
+            // this is done in particular for the checklist_dependency field,
+            // but other fields could use it too, in the future;
+            if (is_array($field['name']) &&
+                isset($field['subfields']) &&
                 is_array($field['subfields']) &&
                 count($field['subfields'])) {
                 foreach ($field['subfields'] as $subfield) {
-                    array_push($relationFields, $subfield);
+                    if (isset($subfield['model']) && $subfield['model'] !== false) {
```

Comment:
Previously all subfields were considered relationship fields and were returned by `getRelationFields()`. Which was wrong, there could have been non-relationship fields there too.

Now, we check the subfields, if they have a model then they're relationship. The same as we do for normal fields.